core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ClusterRepairService.java

```diff
@@ -193,6 +195,21 @@ public FlowIdentifier repairNodes(Long stackId, Set<String> nodeIds, boolean del
         return repairStartResult;
     }
 
+    private boolean isPrimaryGWUnhealthyAndItIsNotSelected(ManualClusterRepairMode repairMode, Set<String> selectedParts, Stack stack) {
```

Comment:
could you add a log line to every branch so it would be easy to tell from the logs which returned and what